java-frontend/src/test/java/org/sonar/java/bytecode/se/BytecodeEGWalkerExecuteTest.java

```diff
@@ -698,6 +698,21 @@ public void test_multianewarray() throws Exception {
     assertThatThrownBy(() -> execute(new Instruction.MultiANewArrayInsn("B", 2))).hasMessage("MULTIANEWARRAY needs 2 values on stack");
   }
 
+  @Test
+  public void test_invoke_dynamic() throws Exception {
+    SymbolicValue lambdaArg = new SymbolicValue();
+    ProgramState programState = execute(new Instruction.InvokeDynamicInsn("(I)Ljava/util/function/Supplier;"), ProgramState.EMPTY_STATE.stackValue(lambdaArg));
+    assertThat(programState.peekValue()).isNotNull();
+    assertThat(programState.peekValue()).isNotEqualTo(lambdaArg);
+
+    programState = execute(new Instruction.InvokeDynamicInsn("()Ljava/util/function/Supplier;"), ProgramState.EMPTY_STATE);
+    assertThat(programState.peekValue()).isNotNull();
+
+    // this is theoretical, in practice lambda will always evaluate to target
```

Comment:
I don't understand this comment : what is target in this context ? What are the following lines testing? edge case ?

Reply:
target is functional interface to which lambda expression is converted, and this in fact can never be void. I decided to remove this part, as this is indeed impossible to happen due to lambda semantic. I also added `NOT_NULL` constraint on the target functional interface, as this is in fact mandated by the spec, see runtime exceptions paragraph in https://docs.oracle.com/javase/specs/jvms/se9/html/jvms-6.html#jvms-6.5.invokedynamic